cmd/bucket-handlers.go

```diff
@@ -495,7 +495,7 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 	vc, _ := globalBucketVersioningSys.Get(bucket)
 	oss := make([]*objSweeper, len(deleteObjectsReq.Objects))
 	for index, object := range deleteObjectsReq.Objects {
-		if apiErrCode := checkRequestAuthType(ctx, r, policy.DeleteObjectAction, bucket, object.ObjectName); apiErrCode != ErrNone {
+		if apiErrCode := checkRequestAuthTypeWithVID(ctx, r, policy.DeleteObjectAction, bucket, object.ObjectName, object.VersionID); apiErrCode != ErrNone {
```

Comment:
Shouldn't we be calling `checkRequestAuthTypeWithVID` in the case `DeleteObjectHandler` too?

Reply:
For DeleteObjectHandler, the VersionID is picked from the request headers in checkRequestAuthType. For multi delete the version id needs to be passed in explicitly because it is in the body

Reply:
Thank you for explaining this.